stingray/pulse/accelsearch.py

```diff
@@ -0,0 +1,283 @@
+from collections.abc import Iterable
+
+import numpy as np
+import scipy
+from scipy import special
+from astropy import log
+from astropy.table import Table
+
+try:
+    from tqdm import tqdm as show_progress
+except ImportError:
+    def show_progress(a):
+        return a
+
+from ..utils import njit, prange
+
+from ..gti import create_gti_mask
+
+
+@njit()
+def pds_from_fft(spectr, nph):
+    return (spectr * spectr.conj()).real * 2 / nph
+
+
+def probability_of_power(level, nbins, n_summed_spectra=1, n_rebin=1):
+    r"""Give the probability of a given power level in PDS.
+
+    Return the probability of a certain power level in a Power Density
+    Spectrum of nbins bins, normalized a la Leahy (1983), based on
+    the 2-dof :math:`{\chi}^2` statistics, corrected for rebinning (n_rebin)
+    and multiple PDS averaging (n_summed_spectra)
+    """
+    try:
+        from scipy import stats
+    except Exception:  # pragma: no cover
+        raise Exception('You need Scipy to use this function')
+
+    epsilon = nbins * stats.chi2.sf(level * n_summed_spectra * n_rebin,
+                                    2 * n_summed_spectra * n_rebin)
```

Comment:
I changed the logic now. I think the current formula is correct, as it uses (1 - CDF^ntrials) instead of ntrials * (1 - CDF). Indeed, the two formulas are almost equivalent when (1 -CDF) * ntrials is very small, but the old formula didn't make sense when this quantity approached 1.

Reply:
New PR has a fully-consistent treatment of ntrials